db/compaction/compaction_service_job.cc

```diff
@@ -404,16 +408,45 @@ Status CompactionServiceCompactionJob::Run() {
   compaction_result_->output_level = compact_->compaction->output_level();
   compaction_result_->output_path = output_path_;
   if (status.ok()) {
-    for (const auto& output_file : sub_compact->GetOutputs()) {
-      auto& meta = output_file.meta;
-      compaction_result_->output_files.emplace_back(
-          MakeTableFileName(meta.fd.GetNumber()), meta.fd.smallest_seqno,
-          meta.fd.largest_seqno, meta.smallest.Encode().ToString(),
-          meta.largest.Encode().ToString(), meta.oldest_ancester_time,
-          meta.file_creation_time, meta.epoch_number, meta.file_checksum,
-          meta.file_checksum_func_name, output_file.validator.GetHash(),
-          meta.marked_for_compaction, meta.unique_id,
-          *output_file.table_properties);
+    if (sub_compact->compaction->SupportsPerKeyPlacement()) {
+      for (const auto& output_file :
+           sub_compact->GetPenultimateLevelOutputs()) {
+        auto& meta = output_file.meta;
+        compaction_result_->output_files.emplace_back(
```

Comment:
I think it would be better to put this in a lambda taking a meta and is_penultimate/is_proximal, and call it in three places (rather than copy-paste in three places)

Or it might be cleaner to add a bool is_penultimate/is_proximal to CompactionOutputs::Output and avoid the new functions in SubcompactionState.